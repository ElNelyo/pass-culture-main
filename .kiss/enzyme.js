// ./node_modules/.bin/jest --env=jsdom ./path/to/file.spec.js --watch
import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = { errors: {} }
      // when
      const wrapper = shallow(<MyComponent {...props} />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
