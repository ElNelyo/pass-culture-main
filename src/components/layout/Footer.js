import classnames from 'classnames'
import { showModal } from 'pass-culture-shared'
import React, { Component } from 'react'
import { connect } from 'react-redux'

import ProfilePicture from './ProfilePicture'
import Menu from '../Menu'

class MenuButton extends Component {
  onClick = () => {
    this.props.showModal(<Menu />, { zIndex: 10002 })
  }

  render() {
    const { borderTop, colored, isFlipped, onTop } = this.props
    const maybeColored = {}
    if (colored) {
      maybeColored.colored = 'colored'
    }
    const style = {}
    if (!colored) {
      style.display = isFlipped ? 'none' : 'block'
    }
    return (
      <footer
        className={classnames('footer', {
          bordered: borderTop,
          colored,
          'on-top': onTop,
        })}
        style={style}
      >
        <div className="button-wrapper">
          <button className="profile-button" onClick={this.onClick} type="button">
            <ProfilePicture alt="Mon menu" {...maybeColored} />
          </button>
        </div>
      </footer>
    )
  }
}

export default connect(
  state => ({ isFlipped: state.verso.isFlipped }),
  {
    showModal,
  }
)(MenuButton)
