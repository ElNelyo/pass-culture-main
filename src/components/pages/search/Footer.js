import React from 'react'

import NavigationFooter from '../../layout/NavigationFooter'

const renderPageFooter = () => (
  <NavigationFooter theme="white" className="dotted-top-red" />
)
export default renderPageFooter
