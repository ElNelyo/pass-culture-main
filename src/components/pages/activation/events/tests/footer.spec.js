// jest --env=jsdom ./src/components/pages/activation/events/tests/footer --watch
import React from 'react'
import { shallow } from 'enzyme'
import Footer from '../footer'

describe('src | components | pages | activation | events | Footer', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // when
      const wrapper = shallow(<Footer />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
