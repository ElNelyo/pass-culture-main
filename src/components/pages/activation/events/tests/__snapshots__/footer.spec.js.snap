// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | activation | events | Footer snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActivationEventsFooter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        title="Continuer sans activer mon pass Culture"
        to="/decouverte"
      >
        <span
          className="fs16 is-white-text"
        >
          Pas maintenant, je jette un oeil d'abord.
        </span>
      </Link>,
      "className": "flex-0 text-center mb24",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <span
          className="fs16 is-white-text"
        >
          Pas maintenant, je jette un oeil d'abord.
        </span>,
        "replace": false,
        "title": "Continuer sans activer mon pass Culture",
        "to": "/decouverte",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Pas maintenant, je jette un oeil d'abord.",
          "className": "fs16 is-white-text",
        },
        "ref": null,
        "rendered": "Pas maintenant, je jette un oeil d'abord.",
        "type": "span",
      },
      "type": [Function],
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          title="Continuer sans activer mon pass Culture"
          to="/decouverte"
        >
          <span
            className="fs16 is-white-text"
          >
            Pas maintenant, je jette un oeil d'abord.
          </span>
        </Link>,
        "className": "flex-0 text-center mb24",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <span
            className="fs16 is-white-text"
          >
            Pas maintenant, je jette un oeil d'abord.
          </span>,
          "replace": false,
          "title": "Continuer sans activer mon pass Culture",
          "to": "/decouverte",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pas maintenant, je jette un oeil d'abord.",
            "className": "fs16 is-white-text",
          },
          "ref": null,
          "rendered": "Pas maintenant, je jette un oeil d'abord.",
          "type": "span",
        },
        "type": [Function],
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
